src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/equalshashcode/InputEqualsHashCodeEqualsParameter.java

```diff
@@ -12,15 +12,15 @@ public boolean equals(String o) {
             return true;
         }
     }
-    public static class TestClass2 { // violation, no correct `equals`
+    public static class TestClass2 {
```

Comment:
This also.

Reply:
This location of comment is wrong, it was ignored before. With this update it becomes actionable and can not stay at this line

Reply:
Why was this missed? Is this connected to the regexp change?

Reply:
Regexp was bad, all such lines were ignored.

Reply:
If this is an issue we ignored such lines, won't we ignore more lines that don't match the regexp? What about `violation` with no comma? Or with comma but nothing afterwards?

Why aren't we finding all violation lines, and then checking if they match a message and if they don't then fail with an unrecognized message.

Reply:
It also sounds like this should be split into 2 commits. 1 on bad regexp, other on severity match which PR is tied to.

Reply:
@SteLeo1602 , ok, this update becomes confusing for others.

Let's create issue on fact that trailing comments without message but with some reason wording is not working. We will fix only it in scope of this issue, separate PR.

And after merging of such update, we will enable this Example.

Reply:
I don't feel any confusion. It seems like there is more going on here and a more deep investigation needs to be done.

We completely missed a `// violation` comment, and that can happen again regardless of what we update regular expression to.

Also this PR was definitely missing explanations on what its doing and why.
https://github.com/checkstyle/checkstyle/pull/15021#discussion_r1670913625
Never got an explanation on this.

Just cause this is tied to an existing issue, doesn't mean we should make a whole bunch of what appears to be unrelated updates without an explanation. If we find bugs, either make them their own PR or tie them to their own issue separate from approved work.

Reply:
@SteLeo1602
new issue is created.
https://github.com/checkstyle/checkstyle/issues/15256

please fix only it in separate PR.